streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -137,8 +141,24 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
     }
     receiverInfo(streamId) = ReceiverInfo(
       streamId, s"${typ}-${streamId}", receiverActor, true, host)
+    logInfo(s"Receiver for stream ${streamId} from ${sender.path.address} " +
+      s"attempts to register.")
+  }
+
+  /** Report a starting receiver */
+  private def startingReceiver(
```

Comment:
But the problem is that receiver restart will deregister the receiver at first, so the actor info of that receiver will be lost, and we need to send `RegisterReceiver` again to re-register itself into the ReceiverTracker. That's why we have to send `RegisterReceiver` again even there's no location or other info changes.